test-poc/framework/src/main/java/org/keycloak/test/framework/server/AbstractKeycloakTestServerSupplier.java

```diff
@@ -22,10 +24,11 @@ public Class<KeycloakIntegrationTest> getAnnotationClass() {
     @Override
     public InstanceWrapper<KeycloakTestServer, KeycloakIntegrationTest> getValue(Registry registry, KeycloakIntegrationTest annotation) {
         KeycloakTestServerConfig serverConfig = SupplierHelpers.getInstance(annotation.config());
+        DatabaseSupplier databaseSupplier = new DevMemDatabaseSupplier();
```

Comment:
This should not create the database supplier, but it should just use the registry to look it up:
```
registry.getDependency(TestDatabase.class, wrapper);
```

See https://github.com/keycloak/keycloak/blob/9247029ca3f1db14d92255e7804dd60a93f10e69/test-poc/framework/src/main/java/org/keycloak/test/framework/realm/RealmSupplier.java#L29-L50 as an example on how to get a dependency in a supplier.

Reply:
Rather then getting supplier I am thinking of fully manged DB called within the KeycloakIntegrationTest annotation as a prerequisite, do you think @stianst it makes sense?